CharlieBackend.Business/Services/LessonService.cs

```diff
@@ -121,6 +122,24 @@ public async Task<Result<IList<LessonDto>>> GetAllLessonsForMentor(long mentorId
             return Result<IList<LessonDto>>.GetSuccess(_mapper.Map<IList<LessonDto>>(lessons));
         }
 
+        public async Task<Result<IList<LessonDto>>> GetLessonsForMentorAsync(FilterLessonsRequestDto filterModel, ClaimsPrincipal userContext)
+        {
+            long accountId = Convert.ToInt32(userContext.Claims.First(x => x.Type.EndsWith("AccountId")).Value);
```

Comment:
Please create a task to write a service like UserSession that will return accounted and other useful data from HttpContext.
Then Inject this service to the constructor and use this service.

Reply:
okay